db/version_builder.cc

```diff
@@ -140,6 +146,57 @@ class VersionBuilder::Rep {
     }
   }
 
+  std::shared_ptr<BlobFileMetaData> GetBlobFileMetaData(
+      uint64_t blob_file_number) const {
+    auto changed_it = changed_blob_files_.find(blob_file_number);
+    if (changed_it != changed_blob_files_.end()) {
+      const auto& meta = changed_it->second;
+      assert(meta);
+
+      return meta;
+    }
+
+    assert(base_vstorage_);
+
+    const auto& base_blob_files = base_vstorage_->GetBlobFiles();
+
+    auto base_it = base_blob_files.find(blob_file_number);
+    if (base_it != base_blob_files.end()) {
+      const auto& meta = base_it->second;
+      assert(meta);
+
+      return meta;
+    }
+
+    return std::shared_ptr<BlobFileMetaData>();
+  }
+
+  Status CheckConsistencyOfOldestBlobFileReference(
+      const VersionStorageInfo* vstorage, uint64_t blob_file_number) const {
+    assert(vstorage);
+
+    // TODO: remove this check once we actually start recoding metadata for
+    // blob files in the MANIFEST.
```

Comment:
Not sure I am following this. What metadata in MANIFEST will make this check unnecessary? By this check, I mean line 180 to 182.

Reply:
Thanks for the review @riversand963 !

Reply:
> Not sure I am following this. What metadata in MANIFEST will make this check unnecessary? By this check, I mean line 180 to 182.

It refers to the blob file metadata I'm in the process of adding. With this patch, the plumbing will be largely in place; however, the logic is currently dormant. On the other hand, those `oldest_blob_file_number` references have been there for a while, so unless I sort of disable these checks for now, all consistency checks would fail when BlobDB is used.

Reply:
Got it. I guess next step will be modifying `LogAndApply` and related callers to write BlobAdd and BlobGarbage to MANIFEST. Since `Recover` uses `VersionBuilder`, I guess that part is easier to update.